### PR TITLE
Added support for handling multiple q parameters

### PR DESCRIPTION
**Pull request by Axle7XStriker**
Currently, APy doesn't support multiple q's as parameters when given for translation. I have resolved this by storing multiple responses(local) first and then sending all at once in a global response. Each parameter q is iteratively translated by a given langpair and an error is raised if at least one of the translation fails.

refs #86

### CODE REVIEW

**Comment by sushain97** [comment]
Hey @Axle7XStriker, do you mind updating this branch with master to fix the build?

**Comment by Axle7XStriker** [comment]
Yeah sure I will do it currently I am on my phone so can't do at the moment. Also, I asked something related to GSoC in one of the closed PR too

> Thanks, also I am working on submitting a proposal to improve the Apertium website. So, do I have to complete all the coding challenges to be eligible? Also for the proposal on the wiki page, there's written that some work has been done in GCI and GSoC previously so do I pick up from where they left work or can I take up other things also, which are mentioned in the issue tracker.
Waiting for reply

**Comment by sushain97** [comment]
> So, do I have to complete all the coding challenges to be eligible?

Nope.

> Also for the proposal on the wiki page, there's written that some work has been done in GCI and GSoC previously so do I pick up from where they left work or can I take up other things also, which are mentioned in the issue tracker.

It would be preferred that you were to complete existing incomplete work prior to picking up new work :) There are lots of features that you can make ready for production to our thousands of users.

**Comment by Axle7XStriker** [comment]
I checked the build and the test case on which it has raised the error seems to be working fine locally. Also, I am not getting how come `self.finish()` is being called twice in one of the tests. Can you please explain to me about the same?

File: servlet.py [hunk 1]
```diff
@@ -501,17 +501,27 @@ def translateAndRespond(self, pair, pipeline, toTranslate, markUnknown, nosplit=
...
+            if pair is not None:
+                pipeline = self.getPipeline(pair)
+                deformat, reformat = self.getFormat()
+                yield self.translateAndRespond(pair,
+                                               pipeline,
+                                               query,
+                                               self.get_argument('markUnknown', default='yes'),
+                                               nosplit=False,
+                                               deformat=deformat, reformat=reformat)
+        if len(self.response) == 1:
```
Comment (sushain97) [inline servlet.py]: This code never executes. `yield` is like `return`. I suggest you read up on how `yield` works and try running this code locally and `print`ing to see what order the statements are executed in.

**Comment by Axle7XStriker** [comment]
OFF-TOPIC: I know it's pretty late to get a review of my proposal but can you please review the proposal provided in the link given below:-
[Apertium Website Improvements](https://docs.google.com/document/d/1_ihitLc5rmnhZzNC4aBkXAb9yDIz0QVAf6DK60S5uBc/edit?usp=sharing)

Waiting for reply.

**Comment by sushain97** [comment]
@Axle7XStriker your proposal looks relatively thorough to me. There are a couple things that I think you could benefit from a little more context on but it's good that you've taken a look at the existing landscape of issues and made a detailed plan based on them. I would be happy to see you continue working on this PR though.

**Comment by xavivars** [comment]
@Axle7XStriker, it would be great if you could also send the proposal to the Apertium mailing list.

**Comment by Axle7XStriker** [comment]
I will surely continue to work on this PR after today.
I have already created a draft of the proposal on the GSoC website. I'll make sure to send it on the mailing list as well.

**Comment by Axle7XStriker** [comment]
Closing this pull request and opening a new one so as to incorporate the latest changes made to the structure repository and other issue related changes.

**Pull request closed without merging**